lib/lightning/attempts.ex

```diff
@@ -232,4 +233,30 @@ defmodule Lightning.Attempts do
   defp adaptor do
     Lightning.Config.attempts_adaptor()
   end
+
+  def mark_unfinished_runs_lost(%Attempt{id: attempt_id}) do
+    # Fetch all runs associated with the attempt
+    runs =
+      Repo.all(
+        from ar in Lightning.AttemptRun,
+          where: ar.attempt_id == ^attempt_id,
+          preload: [:run]
+      )
```

Comment:
This works fine but would use a `join` to avoid a second query:
```
{"SELECT a0.\"id\", a0.\"attempt_id\", a0.\"run_id\", a0.\"inserted_at\" FROM \"attempt_runs\" AS a0 WHERE (a0.\"attempt_id\" = $1)"

and since the database already iterates through the runs, we can filter for the runs where the `exit_reason` is `nil`
```